source/tutorials/openfaas.rst

```diff
@@ -0,0 +1,363 @@
+<https://docs.openvinotoolkit.org/latest/_inference_engine_samples_classification_sample_async_README.html>`_"
+for this function.
+
+
+We'll use the *python3-clearlinux* template as a base and customize it by:
+
+* Adding additional |CL| bundles (*bundles.txt*)
+* Adding additional required python packages (*requirements.txt*)
+* Adding a script to download and convert dldt models (*helper_script.sh*)
+* Finally, we'll develop the python function to be run (*handler.py*)
+
+More ways to customize the |CL| based OpenFaaS templates can be found in the
+`README on GitHub
+<https://github.com/clearlinux/dockerfiles/tree/master/FaaS/OpenFaaS/template#python3-clearlinux>`_.
+
+#. Enter the previously created working directory.
+
+   .. code-block:: bash
+
+      cd ~/faas-example
+
+#. Create a new function skeleton
+
+   .. code-block:: bash
+
+      faas-cli new --lang python3-clearlinux classification-sample --prefix="<your docker name>"
+      
+   This will create the directory  structure below:
+
+   .. code-block:: console
+
+       tree .
+        ├── classification-sample
+        │   ├── bundles.txt
+        │   ├── handler.py
+        │   ├── helper_script.sh
+        │   ├── __init__.py
+        │   └── requirements.txt
+        ├── classification-sample.yml
+
+#. Add the required |CL| bundles to the :file:`bundles.txt` file. 
+
+   .. code-block:: bash
+
+      echo "computer-vision-openvino" >> classification-sample/bundles.txt
+
+#. Add the required python packages to the :file:`requirements.txt` file. 
+
+   .. code-block:: bash
+
+      echo "glob3" >> classification-sample/requirements.txt
+      echo "urllib3" >> classification-sample/requirements.txt
+      echo "networkx==2.3" >> classification-sample/requirements.txt
+
+
+#. OpenCV has a `model downloader and other automation tools
+   <https://github.com/opencv/open_model_zoo/tree/master/tools/downloader>`_
+   to help downloading models and converting them into different formats.
+   Customize the OpenFaaS template to use the model-downloader in the
+   :file:`helper_script.sh` file. The :file:`helper_script.sh` file script
+   gets executed during the build process.
+
+   .. code-block:: bash
+      :caption: helper_script.sh
+      :name: helper_script.sh
+
+      cat classification-sample/helper_script.sh
+      #!/bin/bash
+
+      # Download and convert models
+      export MODEL_DIR="/models"
+      export MO_PATH="/usr/share/openvino/model-optimizer/mo.py"
+      export MODEL_NAME="resnet-50-int8-tf-0001"
+
+      # Download and convert models
+      model-downloader --name $MODEL_NAME -o $MODEL_DIR
+      model-converter --name $MODEL_NAME -d $MODEL_DIR -o $MODEL_DIR --mo $MO_PATH
+
+
+#. With the requirements added to the template. Write a python in the
+   :file:`handler.py` file. This function will parse the input picture URL,
+   find the model path, and call “classification_sample_async” to do image
+   classification. 
+
+   .. code-block:: python
```

Comment:
In running a test build of this PR, I see the python functions need proper indentation below each `def function():` . Will you please apply those changes? I assume the user wants to cut and paste your code-blocks.

Reply:
Thank you, the spacing did indeed get lost. I've updated it.